internal/pkg/service/common/task/orchestrator/orchestrator.go

```diff
@@ -27,95 +25,47 @@ type orchestrator[T any] struct {
 }
 
 func (o orchestrator[T]) start() <-chan error {
-	initDoneOut := make(chan error, 1)
+	initErrCh := make(chan error, 1)
 	o.node.wg.Add(1)
 	go func() {
 		defer o.node.wg.Done()
-		defer o.logger.Info(o.node.ctx, "stopped")
-		b := newRetryBackoff()
+		ctx, span := o.node.tracer.Start(o.node.ctx, spanName, trace.WithAttributes(attribute.String("resource.name", o.config.Name)))
+		stream := o.config.Source.WatchPrefix.GetAllAndWatch(ctx, o.node.client, o.config.Source.WatchEtcdOps...)
+		err := <-stream.SetupConsumer(o.logger.WithComponent("watch.consumer")).
+			WithOnClose(func(err error) {
+				span.End(&err)
+			}).
+			WithForEach(func(events []etcdop.WatchEventT[T], header *etcdop.Header, _ bool) {
+				for _, event := range events {
```

Comment:
Orchestrator was created before the possibility to restart watch streams was added to the `etcdop`.
The code is now easier and there is no duplicity.